src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -39,10 +40,31 @@ object PNode {
 sealed trait PScope extends PNode
 sealed trait PUnorderedScope extends PScope
 
+/**
+ * Contains information about a package
+ *
+ * @param id a unique identifier for the package
+ * @param name the name of the package, does not have to be unique
+ * @param isBuiltIn a flag indicating, if the package comes from within Gobra
+ */
+case class PPackageInfo(id: String, name: String, isBuiltIn: Boolean) extends PNode {
```

Comment:
Me too, but when we want to attach it to the `PPackage` directly, it needs to extend `PNode` because apparently every object in a `PNode` is  assumed to be part of the AST and tried to be cast to a `PNode` which leads to errors if it does not extend it.

Reply:
Weird. Why does it work for `PositionManager`?

Reply:
I did some digging and found the reason why documented in Kiama's `Tree` class: 

```
 * The `originalRoot` value should be the root of a finite, acyclic structure
 * that contains only `Product` instances of type `T` (unless they are skipped,
 * see below).
 *
```

`case classes` are always instances of the type  `Product`. Therefore, `PPackage` may not contain `PackageInfo` without it extending `PNode`. An alternative would be to change `PackageInfo` to a class and implement custom `equals` and `hashCode` methods as we depend on them.